source/extensions/filters/http/local_ratelimit/local_ratelimit.cc

```diff
@@ -94,6 +138,90 @@ Http::FilterHeadersStatus Filter::decodeHeaders(Http::RequestHeaderMap&, bool) {
+    break;
+  case RateLimitFilter::VhRateLimitOptions::Include:
+    populateRateLimitDescriptors(route_entry->virtualHost().rateLimitPolicy(), descriptors,
+                                 route_entry, headers);
+    break;
+  case RateLimitFilter::VhRateLimitOptions::Override:
+    if (route_entry->rateLimitPolicy().empty()) {
+      populateRateLimitDescriptors(route_entry->virtualHost().rateLimitPolicy(), descriptors,
+                                   route_entry, headers);
+    }
+    break;
+  default:
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
+}
+
+void Filter::populateRateLimitDescriptors(const Router::RateLimitPolicy& rate_limit_policy,
+                                          std::vector<RateLimit::Descriptor>& descriptors,
+                                          const Router::RouteEntry* route_entry,
+                                          const Http::HeaderMap& headers) const {
+  for (const Router::RateLimitPolicyEntry& rate_limit :
+       rate_limit_policy.getApplicableRateLimit(config_->stage())) {
+    const std::string& disable_key = rate_limit.disableKey();
+    if (!disable_key.empty() &&
+        !config_->runtime().snapshot().featureEnabled(
+            fmt::format("ratelimit.{}.http_filter_enabled", disable_key), 100)) {
+      continue;
+    }
+    rate_limit.populateDescriptors(*route_entry, descriptors, config_->localInfo().clusterName(),
+                                   headers,
+                                   *decoder_callbacks_->streamInfo().downstreamRemoteAddress(),
+                                   &decoder_callbacks_->streamInfo().dynamicMetadata());
+  }
+}
+
+RateLimitFilter::VhRateLimitOptions
+Filter::getVirtualHostRateLimitOption(const Router::RouteConstSharedPtr& route) {
+  if (route->routeEntry()->includeVirtualHostRateLimits()) {
+    vh_rate_limits_ = RateLimitFilter::VhRateLimitOptions::Include;
+  } else {
+    const auto* specific_per_route_config =
+        Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfig>(
+            "envoy.filters.http.local_ratelimit", route);
+    if (specific_per_route_config != nullptr) {
+      switch (specific_per_route_config->virtualHostRateLimits()) {
+      case envoy::extensions::filters::http::local_ratelimit::v3::
+          LocalRateLimit_VhRateLimitsOptions_INCLUDE:
+        vh_rate_limits_ = RateLimitFilter::VhRateLimitOptions::Include;
+        break;
+      case envoy::extensions::filters::http::local_ratelimit::v3::
+          LocalRateLimit_VhRateLimitsOptions_IGNORE:
+        vh_rate_limits_ = RateLimitFilter::VhRateLimitOptions::Ignore;
+        break;
+      case envoy::extensions::filters::http::local_ratelimit::v3::
+          LocalRateLimit_VhRateLimitsOptions_OVERRIDE:
+      default:
+        vh_rate_limits_ = RateLimitFilter::VhRateLimitOptions::Override;
+      }
+    } else {
+      vh_rate_limits_ = RateLimitFilter::VhRateLimitOptions::Override;
+    }
+  }
+  return vh_rate_limits_;
```

Comment:
we can avoid returning a copy of this, given it's an instance member

Reply:
I actually removed it as an instance member now..